src/Campaigns/Blocks/CampaignGoal/edit.tsx

```diff
@@ -0,0 +1,105 @@
+import {CampaignSelector} from '../shared/components/CampaignSelector';
+
+import './styles.scss';
+
+const getGoalDescription = (goalType: string) => {
+    switch (goalType) {
+        case 'amount':
+            return __('Amount raised', 'give');
+        case 'donations':
+            return __('Number of donations', 'give');
+        case 'donors':
+            return __('Number of donors', 'give');
+        case 'amountFromSubscriptions':
+            return __('Recurring amount raised', 'give');
+        case 'subscriptions':
+            return __('Number of recurring donations', 'give');
+        case 'donorsFromSubscriptions':
+            return __('Number of recurring donors', 'give');
+    }
+}
+
+const getValue = (goalType: string, value: number) => {
+    switch (goalType) {
+        case 'amount':
+        case 'amountFromSubscriptions':
+            return currency.format(value);
+        default:
+            return value;
+    }
+}
+
+const currency = new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: window.GiveCampaignOptions.currency,
+})
+
+/**
+ * @unreleased
+ */
+export default function Edit({attributes, setAttributes}: BlockEditProps<{
+    campaignId: number;
+    goalType: string;
+}>) {
+    const {campaign, hasResolved} = useCampaign(attributes.campaignId);
+
+    const blockProps = useBlockProps();
+
+    const adminBaseUrl = useSelect(
+        // @ts-ignore
+        (select) => select('core').getSite()?.url + '/wp-admin/edit.php?post_type=give_forms&page=give-campaigns',
+        []
+    );
+
+    if (!hasResolved) {
+        return null;
+    }
+
+    return (
+        <div {...blockProps}>
+            <CampaignSelector attributes={attributes} setAttributes={setAttributes}>
+                <div className="give-campaign-goal">
+                    <div className="give-campaign-goal__container">
+                        <div className="give-campaign-goal__container-item">
+                            <span>{getGoalDescription(campaign.goalType)}</span>
+                            <strong>
+                                {getValue(campaign.goalType, campaign?.goalStats?.actual)}
+                            </strong>
+                        </div>
+                        <div className="give-campaign-goal__container-item">
+                            <span>{__('Our goal', 'give')}</span>
+                            <strong>{getValue(campaign.goalType, campaign.goal)}</strong>
+                        </div>
+                    </div>
+                    <div className="give-campaign-goal__progress-bar">
+                        <div className="give-campaign-goal__progress-bar-container">
+                            <div className="give-campaign-goal__progress-bar-progress"
+                                 style={{width: `${campaign?.goalStats?.percentage}%`}}>
+                            </div>
+                        </div>
+                    </div>
+                </div>
```

Comment:
Honest question: why not use the server side rendered content here?

Reply:
Yeah, I thought that myself. Let me refactor that

Reply:
Resolved https://github.com/impress-org/givewp/pull/7702/commits/0926bad9c9a7a234e7ed3ed19c7813241d591d4a